src/einsteinpy/symbolic/tensor.py

```diff
@@ -116,6 +116,29 @@ def tensor_product(tensor1, tensor2, i=None, j=None):
     )
 
 
+def _parse_vector(x, syms):
+    # make a proper vector (tensor of config 'u') out of given x
+    if isinstance(x, BaseRelativityTensor):
+        if x.config == "u" and x.syms == syms:
+            return x
+        raise ValueError("Expected a tensor of config 'u' with syms %s" % syms)
+
```

Comment:
As the test suggests, the problem is about the number of execution paths through code. So, maybe we should keep this function simple and directly do not enter in `_parse_vector` if `x` is already a `BaseRelativityTensor`,` x.config == "u" ` ... , what I suggest is make this check in `def Derivative()` (line 504). 

So I will delete this from `_parse_vector` :
```
if isinstance(x, BaseRelativityTensor):
        if x.config == "u" and x.syms == syms:
            return x
        raise ValueError("Expected a tensor of config 'u' with syms %s" % syms)
```
Let me know if it works!